File: app/src/main/java/com/example/breezil/favnews/di/module/MainFragmentsModule.java
package com.example.breezil.favnews.di.module;

import com.example.breezil.favnews.ui.BookMarksFragment;
import com.example.breezil.favnews.ui.HeadlinesFragment;
import com.example.breezil.favnews.ui.PreferredFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class MainFragmentsModule {
    @ContributesAndroidInjector
    abstract PreferredFragment contributePreferredFragment();

    @ContributesAndroidInjector
    abstract HeadlinesFragment contributeHeadlinesFragment();

    @ContributesAndroidInjector
    abstract BookMarksFragment contributeBookMarksFragment();
}

File: app/src/main/java/com/example/breezil/fave/di/module/MainFragmentsModule.kt
package com.example.breezil.fave.di.module

import com.example.breezil.fave.ui.main.fragment.BookMarksFragment
import com.example.breezil.fave.ui.main.fragment.HeadlinesFragment
import com.example.breezil.fave.ui.main.fragment.PreferredFragment

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class MainFragmentsModule {
    @ContributesAndroidInjector
    internal abstract fun contributePreferredFragment(): PreferredFragment

    @ContributesAndroidInjector
    internal abstract fun contributeHeadlinesFragment(): HeadlinesFragment

    @ContributesAndroidInjector
    internal abstract fun contributeBookMarksFragment(): BookMarksFragment
}
